src/Core/Scheduler/Scheduler.js

```diff
@@ -119,11 +119,20 @@ Scheduler.prototype.runCommand = function runCommand(command, queue, executingCo
 Scheduler.prototype.execute = function execute(command) {
     // TODO: check for mandatory commands fields
 
-
     // parse host
     const layer = command.layer;
+    let host;
 
-    const host = layer.url ? new URL(layer.url).host : undefined;
+    if (layer.url) {
+        if (layer.url.includes('://')) {
+            host = new URL(layer.url).host;
+        } else {
+            // local file
```

Comment:
the problem is not with local files (`file://local/path` works fine with the old code), but with relative path (without scheme). Can you update your commit message reflecting that please? 

Btw in this message, let's not use the word "problem", when there's a fix, it's obvious there was a problem ;-)  I would leave out the `:` too and the second `fix`, in order to make it readable aloud. Maybe simply `Fix getting host with relative url in scheduler` will do?

Reply:
IMHO the `fix:` prefix is useful (for example see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#type)

But you could reword the part after `:` to avoid the repetition.
And add a description in your commit message, it's more important than the PR text (hint: github could close tomorrow and all PR text will be lost but the git commit message would remain).

```
fix (scheduler): improve layers with relative url support

Prior to this commit using a relative url (like ./foo.gpx) would produce 
an error.
This is fixed by this commit by using the optional 'base' parameter of URL
 constructor that specifies the base address if the first parameter is a 
relative URL.
(see https://developer.mozilla.org/en-US/docs/Web/API/URL/URL)

```

Reply:
updated